src/server.py

```diff
@@ -298,22 +304,28 @@ def terminate_sound_for_sender(self, uuid_, connection, sender, invocation,
                              "not exist".format(uuid_))
         elif (uuid_ in self.registry.sounds and
               (uuid_ not in self.registry.refcount or
-               sender not in self.registry.refcount[uuid_])):
+               sender != self.get_sound(uuid_).bus_name)):
             self.logger.info("Sound {} was supposed to be "
                              "refcounted by the bus, name \'{}\' but "
                              "it wasn\'t.".format(uuid_, sender))
         else:
             if uuid_ in self.registry.sounds:
-                self.unref_on_stop(uuid_, sender, term_sound)
+                # Stop by UUID.
+                self.unref_on_stop(self.get_sound(uuid_), term_sound)
             elif uuid_ in self.registry.uuids_by_event_id:
+                # Stop by sound event id.
                 sound_event_id = uuid_
                 for uuid_ in self.registry.uuids_by_event_id[sound_event_id]:
-                    self.unref_on_stop(uuid_, sender, term_sound)
+                    sound = self.get_sound(uuid_)
```

Comment:
As I told in some conversation with you, a sound can be stopped now by its sound event id (not only by its UUID).

In line 316, the uuid_ is actually the sound_event_id. Then in line 318 we iterate through all hte uuids for that sound event id, so here they are uuids.

Yes... I see now this is confusing. Maybe it would be a good idea to rename `uuid_` to `uuid_or_event_id` in the method `terminate_sound_for_sender`. Any suggestion here?